### PR TITLE
Refactor(eos_designs): Optimize eos_designs_structured_config file write

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Optimize eos_designs_structured_config file write

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

- Use Python native file operations instead of Ansible Copy module.

## How to test

Tested locally with both yml, yaml and json files.
- [x] Changed true/false correct
- [x] File mode set correctly 
- [x] Performance improvement for small configs ~ 2x. For very large configs ~ 10%

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/plugins/action/eos_designs_structured_config.py [hunk 1]
```diff
@@ -145,19 +149,24 @@ def run(self, tmp=None, task_vars=None):
             with self._templar.set_temporary_context(available_variables=template_vars):
                 output = self._templar.template(output, fail_on_undefined=False)
 
-        # If the argument 'dest' is set, write the output data to a file.
-        if self.dest:
-            # Depending on the file suffix of 'dest' (default: 'json') we will format the data to yaml or just write the output data directly.
+        # If the argument 'dest' (filename) is set, write the output data to a file.
+        if filename:
+            # Depending on the file suffix of 'filename' (default: 'json') we will format the data to yaml or just write the output data directly.
             # The Copy module used in 'write_file' will convert the output data to json automatically.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/plugins/action/eos_designs_structured_config.py]: Remove this comment
```suggestion
```

File: ansible_collections/arista/avd/plugins/action/eos_designs_structured_config.py [hunk 2]
```diff
@@ -171,26 +180,29 @@ def run(self, tmp=None, task_vars=None):
...
+        path = Path(filename)
+        if path.exists():
+            if path.read_text(encoding="UTF-8") == content:
+                return False
+
+        else:
+            # Create parent dirs automatically.
+            path.parent.mkdir(mode=int(dir_mode, 8), parents=True, exist_ok=True)
+            # Touch file
+            path.touch(mode=int(file_mode, 8))
```
Comment (gmuloc) [inline ansible_collections/arista/avd/plugins/action/eos_designs_structured_config.py]:
```suggestion
        if not path.exists():
            # Create parent dirs automatically.
            path.parent.mkdir(mode=int(dir_mode, 8), parents=True, exist_ok=True)
            # Touch file
            path.touch(mode=int(file_mode, 8))
        elif path.read_text(encoding="UTF-8") == content:
            return False
```

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/action/eos_designs_structured_config.py]: Hey I copied your code :)

File: ansible_collections/arista/avd/plugins/action/eos_designs_structured_config.py [hunk 3]
```diff
@@ -171,27 +177,27 @@ def run(self, tmp: Any = None, task_vars: dict | None = None) -> dict:
...
-            templar=self._templar,
-            shared_loader_obj=self._shared_loader_obj,
-        )
+        Parameters
+        ----------
+            content: The content to write
+            filename: Target filename
 
-        return copy_action.run(task_vars=task_vars)
+        Returns
```
Comment (Vibhu-gslab) [inline ansible_collections/arista/avd/plugins/action/eos_designs_structured_config.py]: This will clear CI.
```suggestion
        Returns:
```

**Review by MaheshGSLAB — APPROVED** [review]

**Review by Shivani-gslab — APPROVED** [review]

**Pull request merged**